var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -6,50 +6,156 @@
 from spack import *
 
 
-class Hiop(CMakePackage):
+class Hiop(CMakePackage, CudaPackage):
     """HiOp is an optimization solver for solving certain mathematical
     optimization problems expressed as nonlinear programming problems.
     HiOp is a lightweight HPC solver that leverages application's existing
     data parallelism to parallelize the optimization iterations by using
     specialized linear algebra kernels."""
 
     homepage = "https://github.com/LLNL/hiop"
-    git      = "https://github.com/LLNL/hiop.git"
+    git = "https://github.com/LLNL/hiop.git"
 
+    # Most recent tagged snapshot is the preferred version when profiling.
+    version('0.3.99.3', tag='v0.3.99.3', preferred=True)
+    version('0.3.99.2', tag='v0.3.99.2')
+    version('0.3.99.1', tag='v0.3.99.1')
+    version('0.3.99.0', tag='v0.3.99.0')
     version('0.1', tag='v0.1')
+    version('0.2', tag='v0.2')
+    version('0.3', tag='v0.3')
 
-    variant('mpi', default=True,
-            description='Enable/Disable MPI')
+    # Development branches
+    version('master', branch='master')
+    version('develop', branch='develop')
 
+    variant(
+        'jsrun',
+        default=False,
+        description='Enable/Disable jsrun command for testing')
+    variant(
+        'shared',
+        default=False,
+        description='Enable/Disable shared libraries')
+    variant('mpi', default=True, description='Enable/Disable MPI')
+    variant('raja', default=False, description='Enable/Disable RAJA')
+    variant('kron', default=False, description='Enable/Disable Kron reduction')
+    variant(
+        'sparse',
+        default=False,
+        description='Enable/Disable Sparse linear algebra')
     variant('deepchecking', default=True,
-            description='Ultra safety checks - \
-            used for increased robustness and self-diagnostics')
+            description='Ultra safety checks - '
+            'used for increased robustness and self-diagnostics')
 
-    depends_on('mpi', when='+mpi')
     depends_on('lapack')
     depends_on('blas')
+    depends_on('cmake@3.15:', type='build')
```

Comment:
You might want to be using cmake 3.18+ since you're using https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_ARCHITECTURES.html further below